overlord/restart/restart.go

```diff
@@ -362,3 +362,138 @@ func ReplaceBootID(st *state.State, bootID string) {
 	rm := restartManager(st, "internal error: cannot mock a restart request before RestartManager initialization")
 	rm.bootID = bootID
 }
+
+// markTaskForRestart sets certain properties on a task to mark it for system restart.
+// The status argument is the status that the task will have after the system restart.
+func markTaskForRestart(t *state.Task, status state.Status) {
+	// Preserve previous restart behaviour for classic in the undo cases.
```

Comment:
maybe this should have a XXX, I'm not entirely sure whether given the overall new approach this decision makes still sense. Maybe @alfonsosanchezbeato also has thoughts on this

Reply:
added XXX for this, would be nice if this isn't actually neccessary anymore

Reply:
If there is some check for classic in the logic that finally asks the system to restart, maybe. But it would need to check if the task was in Undo/Undone, so that might make things more difficult I would say. So I think it will be better to keep this here at least for the moment.